### PR TITLE
Fix missing parent constructor call and get_class usage in GenericEventWrapper

### PR DESCRIPTION
**Pull request by ChristophWurst**
* The parent constructor was not called
* `get_class` does not allow null values in php7.2+

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: lib/private/EventDispatcher/GenericEventWrapper.php [hunk 1]
```diff
@@ -40,24 +40,26 @@ class GenericEventWrapper extends GenericEvent {
...
 		$this->logger = $logger;
 		$this->event = $event;
 		$this->eventName = $eventName;
 	}
 
 	private function log() {
+		$class = ($this->event !== null && is_object($this->event)) ? get_class($this->event) : 'null';
 		$this->logger->info(
 			'Deprecated event type for {name}: {class} is used',
-			[ 'name' => $this->eventName, 'class' => is_object($this->event) ? get_class($this->event) : 'null' ]
```
Comment (kesselb) [inline lib/private/EventDispatcher/GenericEventWrapper.php:51]: What was wrong with the check? https://3v4l.org/rfOnp :thinking:

**Review by rullzer — APPROVED** [review]
sure

File: lib/private/EventDispatcher/GenericEventWrapper.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/private/EventDispatcher/GenericEventWrapper.php:51]: my php inspection inside phpstorm suggested this :shrug: 

Bildschirmfoto von 2020-06-08 11-20-54

I did no think much about it but just changed it to a more robust check :wink:

Comment (kesselb) [reply lib/private/EventDispatcher/GenericEventWrapper.php:51]: Ah. So `is_object` is not detected as not null check.

**Pull request merged**